test-integration/filter-balancer.js

```diff
@@ -0,0 +1,83 @@
+/* global artifacts */
+
+const utils = require("../utils/utilities.js");
+const { deployArgent } = require("../utils/argent-deployer.js");
+
+const BPool = artifacts.require("BPool");
+const BalancerFilter = artifacts.require("BalancerFilter");
+
+const amount = web3.utils.toWei("1");
+
+contract("Balancer Filter", (accounts) => {
+  let argent;
+  let wallet;
+  let pool;
+
+  before(async () => {
+    argent = await deployArgent(accounts);
+
+    pool = await BPool.at("0x1eff8af5d577060ba4ac8a29a13525bb0ee2a3d5");
+    const filter = await BalancerFilter.new();
+    await argent.dappRegistry.addDapp(0, pool.address, filter.address);
+  });
+
+  beforeEach(async () => {
+    wallet = await argent.createFundedWallet({ WETH: amount });
+  });
+
+  const deposit = async () => argent.multiCall(wallet, [
```

Comment:
We could check that the balances have been updated as part of the `deposit` and `withdraw` methods.